### PR TITLE
Add support for ngx_http_api_module

### PR DESCRIPTION
**Pull request by mariomac**
#### Description of the changes

- Support for `ngx_http_api_module`.
- New configuration options:
    - `connection_timeout`: timeout (in seconds) for the connection from the integration to Nginx
    - `status_module` (default: `discover`). Accepted values:
        * `ngx_http_stub_status_module`
        * `ngx_http_status_module`
        * `ngx_http_api_module`
        * `discover` to automatically choose between `ngx_http_stub_status_module` or `ngx_http_status_module` (added for backwards-compatibility)
    - `endpoints`: if `status_module` is `ngx_http_api_module`, comma separated list, NON PARAMETERIZED, Endpoints   

### Author

- [x] add a risk label after carefully considering the "blast radius" of your changes
- [x] describe the _intent_ of your changes in the description. don't just rewrite your code in prose
- [x] assign at least one reviewer

### Reviewer

- [ ] review code for readability
- [ ] verify that high risk behavior changes are well tested
- [ ] check license for any new external dependency
- [ ] ask questions about anything that isn't clear and obvious
- [ ] approve the PR when you consider it's good to merge

### CODE REVIEW

File: nginx-config.yml.sample [hunk 1]
```diff
@@ -4,7 +4,14 @@ instances:
     - name: nginx-server-metrics
       command: metrics
       arguments:
+          # If you're using ngx_http_api_module be certain to use the full path up to and including the version number
           status_url: http://127.0.0.1/status
+	      # Name of Nginx status module OHI is to query against. discovery | ngx_http_stub_status_module | ngx_http_status_module | ngx_http_api_module
+	      status_module: discovery
```
Comment (cristianciutea) [inline nginx-config.yml.sample]: is this line proper formatted?

File: src/nginx.go [hunk 2]
```diff
@@ -2,33 +2,42 @@ package main
...
 type argumentList struct {
 	sdk_args.DefaultArgumentList
-	StatusURL        string `default:"http://127.0.0.1/status" help:"NGINX status URL."`
-	ConfigPath       string `default:"/etc/nginx/nginx.conf" help:"NGINX configuration file."`
-	RemoteMonitoring bool   `default:"false" help:"Identifies the monitored entity as 'remote'. In doubt: set to true."`
+	StatusURL         string `default:"http://127.0.0.1/status" help:"NGINX status URL. If you are using ngx_http_api_module be sure to include the full path ending with the API version number"`
+	ConfigPath        string `default:"/etc/nginx/nginx.conf" help:"NGINX configuration file."`
+	RemoteMonitoring  bool   `default:"false" help:"Identifies the monitored entity as 'remote'. In doubt: set to true."`
+	ConnectionTimeout int    `default:"1" help:"OHI connection to Nginx timeout in seconds"`
+	StatusModule      string `default:"discover" help:"Name of Nginx status module. discover | ngx_http_stub_status_module | ngx_http_status_module | ngx_http_api_module"`
```
Comment (cristianciutea) [inline src/nginx.go]: is 'discover' the correct default value?

File: nginx-config.yml.sample (hunk 1 above)
Comment (mariomac) [reply nginx-config.yml.sample]: Yes

File: src/nginx.go (hunk 2 above)
Comment (mariomac) [reply src/nginx.go]: Yes, it keeps the previous behaviour to avoid breaking changes

File: nginx-config.yml.sample (hunk 1 above)
Comment (scellef) [reply nginx-config.yml.sample]: There are tabstop characters on those two lines, causing a YAML syntax error when the agent tries to parse the config.

**Pull request merged**